nixos/modules/services/x11/dunst.nix

```diff
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+   cfg = config.services.dunst;
+in {
+
+options.services.dunst = {
+
+  enable = mkEnableOption "the dunst notifications daemon";
+
+  iconDirs = mkOption {
+    type = with types; listOf path;
+    default = [];
+    example = literalExample ''
+      [ "''${pkgs.gnome3.adwaita-icon-theme}/share/icons/Adwaita/48x48" ]
+    '';
+    description = ''
+      Paths to icon folders.
+    '';
+  };
+
+  extraCliOptions = mkOption {
+    type = types.str;
+    default = "";
+    description = ''
+      This gets appended verbatim to the command line of dunst.
+    '';
+  };
+
+  globalConfig = mkOption {
+    type = with types; nullOr (attrsOf str);
+    default = null;
+    description = ''
+      Set the global configuration section for dunst.
+    '';
+  };
+
+  shortcutConfig = mkOption {
+    type = with types; nullOr (attrsOf str);
+    default = null;
+    description = ''
+      Set the shortcut configuration for dunst.
+    '';
+  };
+
+  urgencyConfig = {
+    low = mkOption {
+      type = with types; nullOr (attrsOf str);
+      default = null;
+      description = ''
+        Set the low urgency section of the dunst configuration.
+      '';
+    };
+    normal = mkOption {
+      type = with types; nullOr (attrsOf str);
+      default = null;
+      description = ''
+        Set the normal urgency section of the dunst configuration.
+      '';
+    };
+    critical = mkOption {
+      type = with types; nullOr (attrsOf str);
+      default = null;
+      description = ''
+        Set the critical urgency section of the dunst configuration.
+      '';
+    };
+  };
+
+  rules = mkOption {
+    type = with types; attrsOf (attrsOf str);
+    default = {};
+    description = ''
+       Rules allow the conditional modification of notifications.
+
+       Note that rule names may not be one of the following
+       keywords already used internally:
+         'global' 'experimental' 'frame' 'shortcuts',
+         'urgency_low' 'urgency_normal' 'urgency_critical'
+
+       There are 2 parts in configuring a rule: Defining when a rule
+       matches and should apply (called filtering in the man page)
+       and then the actions that should be taken when the rule is
+       matched (called modifying in the man page).
+    '';
+    example = {
+      signed_off = {
```

Comment:
To make newlines show up nicely in the manual, you need
```nix
{
  example = literalExample ''
    ...
  '';
}
```

Reply:
Ok, I am not sure I understand that one. Which newlines exactly / can you clarify that?

Reply:
Like this:

```nix
mkOption {
  example = literalExample ''
    {
      signed_off = {
        appname = "Pidgin";
        # ...
      };
    }
  '';
}
```

If a Nix value that spans multiple lines is assigned to `example` directly, it will render as a single line in the manual, which is hard to read